persist/fs/read.go

```diff
@@ -120,21 +121,57 @@ func NewReader(
 	}, nil
 }
 
-func (r *reader) Open(namespace ident.ID, shard uint32, blockStart time.Time) error {
-	var err error
+func (r *reader) Open(opts ReaderOpenOptions) error {
+	var (
+		namespace     = opts.Identifier.Namespace
+		shard         = opts.Identifier.Shard
+		blockStart    = opts.Identifier.BlockStart
+		snapshotIndex = opts.Identifier.Index
+		err           error
+	)
+
+	var (
+		shardDir            string
+		checkpointFilepath  string
+		infoFilepath        string
+		digestFilepath      string
+		bloomFilterFilepath string
+		indexFilepath       string
+		dataFilepath        string
+	)
+	switch opts.FilesetType {
+	case persist.FilesetSnapshotType:
+		shardDir = ShardSnapshotsDirPath(r.filePathPrefix, namespace, shard)
+		checkpointFilepath = snapshotPathFromTimeAndIndex(shardDir, blockStart, checkpointFileSuffix, snapshotIndex)
```

Comment:
how about using something like the following:

```
// persist/fs/read.go
func (r *reader) Open(...) error {
  var (
		namespace     = opts.Identifier.Namespace
		shard         = opts.Identifier.Shard
		blockStart    = opts.Identifier.BlockStart
		snapshotIndex = opts.Identifier.Index
		err           error
		filePaths := make(map[string]string, len(filesetFileSuffixes))
	)
  switch opts.FilesetType { 
  case persist.FilesetSnapshotType:
    shardDir := ShardSnapshotsDirPath(r.filePathPrefix, namespace, shard)
    for _, suffix := range filesetFileSuffixes {
      filePaths[suffix] = snapshotPathFromTimeAndIndex(shardDir, blockStart, suffix, snapshotIndex)
    }
  case persist.FilesetFlushType:
    shardDir = ShardDataDirPath(r.filePathPrefix, namespace, shard)
    for _, suffix := range filesetFileSuffixes {
      filePaths[suffix] = filesetPathFromTime(shardDir, blockStart, suffix)
    }
  default:
  // ...
  }
  
  if err := r.readCheckpointFile(filePaths[checkpointFileSuffix], blockStart); err != nil {
  // ...
  } 
  // ... 
}
 
```

```
// persist/fs/fs.go
const (
const (
	infoFileSuffix        = "info"
	indexFileSuffix       = "index"
	summariesFileSuffix   = "summaries"
...
)

var (
  filesetFileSuffixes = []string { infoFileSuffix, indexFileSuffix, summariesFileSuffix, bloomFilterFileSuffix, dataFileSuffix, digestFileSuffix, checkpointFileSuffix }
)
```

Reply:
ehh I don't find this to be any cleaner, if anything its more confusing cause you have to track down the map and slice logic. I prefer to leave this explicit